third_party/move/move-vm/runtime/src/move_vm.rs

```diff
@@ -71,20 +71,20 @@ impl MoveVM {
     ///     and apply the effects to the storage when the Session ends.
     pub fn new_session<'r>(
         &self,
-        remote: &'r impl MoveResolver<PartialVMError>,
+        remote: &'r impl MoveResolver<Arc<CompiledModule>, PartialVMError>,
     ) -> Session<'r, '_> {
         self.new_session_with_extensions(remote, NativeContextExtensions::default())
     }
 
     /// Create a new session, as in `new_session`, but provide native context extensions.
     pub fn new_session_with_extensions<'r>(
         &self,
-        remote: &'r impl MoveResolver<PartialVMError>,
+        remote: &'r impl MoveResolver<Arc<CompiledModule>, PartialVMError>,
```

Comment:
With the current changes, we are pretty much forcing the MoveResolver to have the associated type argument Arc<CompiledModule>. Should we just remove the type argument in this case?

Reply:
Yes, If only VM was generic we could use it, but good point, here it is always arced module, let's just use the concrete type

Reply:
Moved to move-vm-types, removed associated types